int32 keypoint_num
float32 repro_error